### PR TITLE
Parse architecture from oval file

### PR DESCRIPTION
**Pull request by Allda**
Rpm package architecture is parsed from oval files and stored in a vuln database table.

Architecture value depends on architecture operation (equal, not-equal, pattern match, ...) so operation is also stored in DB as an enum. This can be later used during DB query to select only the right architecture.

### CODE REVIEW

**Comment by ldelossa** [comment]
@Allda some merge conflicts.

**Comment by Allda** [comment]
@ldelossa it's ready for review

File: package.go [hunk 1]
```diff
@@ -26,3 +26,8 @@ type Package struct {
 	// Package architecture
 	Arch string `json:"arch,omitempty"`
 }
+
+const (
```
Comment (ldelossa) [inline package.go:30]: :+1:

File: vulnerability.go [hunk 2]
```diff
@@ -31,5 +60,6 @@ type Vulnerability struct {
 	// a string specifying the package version the fix was relased in
 	FixedInVersion string `json:"fixed_in_version"`
 	// Range describes the range of versions that are vulnerable.
-	Range *Range `json:"range,omitempty"`
+	Range         *Range `json:"range,omitempty"`
+	ArchOperation ArchOp `json:"arch_op,omitempty"`
```
Comment (ldelossa) [inline vulnerability.go:64]: Comment this.

**Pull request merged**